lib/Sema/ConstraintGraph.cpp

```diff
@@ -289,52 +311,51 @@ static bool isUsefulForReferencedVars(Constraint *constraint) {
   }
 }
 
-void ConstraintGraphNode::introduceToInference(Constraint *constraint,
-                                               bool notifyReferencedVars) {
+void ConstraintGraphNode::introduceToInference(Constraint *constraint) {
   if (forRepresentativeVar()) {
     auto fixedType = TypeVar->getImpl().getFixedType(/*record=*/nullptr);
     if (!fixedType)
       getCurrentBindings().infer(constraint);
   } else {
     auto *repr =
         getTypeVariable()->getImpl().getRepresentative(/*record=*/nullptr);
-    CG[repr].introduceToInference(constraint, /*notifyReferencedVars=*/false);
+    CG[repr].introduceToInference(constraint);
   }
 
+  /*
   if (!notifyReferencedVars || !isUsefulForReferencedVars(constraint))
     return;
 
-  for (auto *fixedBinding : getReferencedVars()) {
-    CG[fixedBinding].introduceToInference(constraint,
-                                          /*notifyReferencedVars=*/false);
-  }
+  this->notifyReferencedVars([&](ConstraintGraphNode &referencedVar) {
+    referencedVar.introduceToInference(constraint);
+  });
+*/
 }
 
-void ConstraintGraphNode::retractFromInference(Constraint *constraint,
-                                               bool notifyReferencedVars) {
+void ConstraintGraphNode::retractFromInference(Constraint *constraint) {
   if (forRepresentativeVar()) {
     auto fixedType = TypeVar->getImpl().getFixedType(/*record=*/nullptr);
     if (!fixedType)
       getCurrentBindings().retract(constraint);
   } else {
     auto *repr =
         getTypeVariable()->getImpl().getRepresentative(/*record=*/nullptr);
-    CG[repr].retractFromInference(constraint, /*notifyReferencedVars=*/false);
+    CG[repr].retractFromInference(constraint);
   }
 
+  /*
```

Comment:
Hey @xedin :)
We end up with this commented, is that correct?

Reply:
Apparently I forgot to push a commit that cleans that up...

Reply:
https://github.com/apple/swift/pull/37841